config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -169,6 +209,43 @@ private Optional<ConfigDocument> getLatestVersionConfigDoc(
     return Optional.empty();
   }
 
+  private Map<ConfigResourceContext, Optional<ConfigDocument>> getLatestVersionConfigDocs(
+      Set<ConfigResourceContext> configResourceContexts) throws IOException {
+    // build filter
+    Filter[] childFilters = new Filter[configResourceContexts.size()];
+    int index = 0;
+    for (ConfigResourceContext configResourceContext : configResourceContexts) {
+      childFilters[index] = getConfigResourceContextFilter(configResourceContext);
+      index += 1;
+    }
+    Filter filter = new Filter();
+    filter.setOp(Filter.Op.OR);
```

Comment:
Do we have IN filter support? Could probably do this as a single filter - here, we're making a pretty bulky tree - each one of these filters includes multiple AND'd filters that are duplicated for each context (tenant, resource, namespace).

Reply:
I think we would only be duplicating tenant right? Because each of the configs to be upserted could have a different resource and namespace.
```
message UpsertAllConfigsRequest {
  repeated ConfigToUpsert configs = 1;

  message ConfigToUpsert {
    string resource_name = 1;
    string resource_namespace = 2;
    string context = 3;
    google.protobuf.Value config = 4;
  }
}
```